osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/tasks/EventCleanupJob.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.domain.core.application.tasks;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.joda.time.DateTime;
+import org.opensmartgridplatform.adapter.domain.core.application.services.TransactionalEventService;
+import org.opensmartgridplatform.domain.core.entities.Event;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.util.Assert;
+
+/**
+ * Scheduled job which will find {@link Event} records older than
+ * {@link EventCleanupJob#eventRetentionPeriodInMonths}}. The old events are
+ * converted to CSV file and stored to the file system. Then the old events are
+ * deleted from the database.
+ */
+@DisallowConcurrentExecution
+public class EventCleanupJob extends CsvWriterJob implements Job {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventCleanupJob.class);
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.retention}")
+    private int eventRetentionPeriodInMonths;
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.csv.file.location}")
+    private String csvFileLocation;
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.csv.file.prefix}")
+    private String csvFilePrefix;
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.csv.file.compression.enabled}")
+    private boolean csvFileCompressionEnabled;
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.page.size}")
+    private int eventPageSize;
+
+    @Autowired
+    private TransactionalEventService transactionalEventService;
+
+    @Override
+    public void execute(final JobExecutionContext context) throws JobExecutionException {
+        LOGGER.info("Quartz triggered cleanup of database - event records.");
+        final DateTime start = DateTime.now();
+
+        final Date retention = this.calculateDate();
+        final List<Event> oldEvents = this.transactionalEventService.getEventsBeforeDate(retention, this.eventPageSize);
+        if (!oldEvents.isEmpty()) {
+            this.saveEventsToCsvFile(oldEvents);
+
+            LOGGER.info("Deleting events...");
+            this.transactionalEventService.deleteEvents(oldEvents);
+
+            System.gc();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Don't try to be smarter than the JVM, remove this call to run the garbage collector. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1215)